packages/python-django-storages/python-django-storages.spec

```
@@ -32,7 +32,39 @@ Obsoletes:      python3.11-%{pypi_name} < %{version}-%{release}
 %description
 %{summary}
 
+%package -n python%{python3_pkgversion}-%{pypi_name}+s3
+Summary: Metapackage for python%{python3_pkgversion}-pulpcore: s3 compatible extra
+Requires: python%{python3_pkgversion}-boto3 >= 1.4.4
 
+%description -n python%{python3_pkgversion}-%{pypi_name}+s3
+This is a metapackage bringing in filecache extra requires for python%{python3_pkgversion}-%{pypi_name}
+It contains no code, just makes sure the dependencies are installed.
+
+%files -n python%{python3_pkgversion}-%{pypi_name}+s3
+%ghost %{python3_sitelib}/%{src_name}-%{version}.dist-info/
```

Why is this a ghost? Looks like it's included
```suggestion
%{python3_sitelib}/%{src_name}-%{version}.dist-info/
```

---

To get the extra packages with rich dependencies, we need the `%ghost` macro:

```
The dependency generator for extras activates if the following holds:
The package name must end with +EXTRA (where EXTRA is the extra name).
The package must contain the .dist-info directory, usually as %ghost.
```

https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#Extras

---

> The package must contain the .dist-info directory, usually as %ghost.

Oh, so this is some dark magic?

---

Yes it's, it took me 2 or 3 days to understand why my packages needed the `ghost` macro

---

Then I'm glad I asked then :)